### ISSUE TITLE
Problem with streaming avi files

### ISSUE THREAD

**Issue by mcarluec** [ampache/ampache#1577]
Hello,
I can't stream my avi files or mkv files. Only mp4 are recognized.
It's probably a config on the ampache.cfg.php... but I have not find what.
Someone can help me ?
Thank

# Environment

 * Ampache version: 3.8.2
 * Server operating system: yunohost 2.6.4
 * Client operating system: Firefox 54.0.1 (64 bits)

**Comment by wagnered** [comment]
@mcarluec,

I assume you are referring to the webplayer as I could not duplicate
your problem by actually streaming the videos to my system and playing
both mkv and avi files with my local video player.

The html5 player, I believe, still plays only mp4 and ogg videos.  Adobe
Flash plays only swf and mp4 files.

It is necessary to experiment with the transcoding options in
ampache.cfg.php.I experimented a little but couldn't find the correct
options.

As a last resort, the avi's can be converted with ffmpeg:

    /usr/bin/ffmpeg -i input.avi  output.mp4


fyi: Ampache 3.8.3 release is now available.

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by mcarluec** [comment]
Thank you for your quickly answer.
It's exact that I'm speaking of webplayer html5...
When you said to experiment with the transcoding options. Do you mind that I must create a encode_args_avi in ampache.cfg.php ?
I have a lot of avi files and I don't want convert them in mp4... Too long!

**Comment by mcarluec** [comment]
Yes !
I have added a line encode_args_avi with the arguments like encode_args_webm and pass transcode_avi from "allowed" to "required" and it's OK !
But my server maybe  isn't very strong because the streaming is intermittent.

**Comment by wagnered** [comment]
Yes, it takes a lot of cpu processing power to transcode a file on the
fly.  If you are watching the video from a web browser on the server,
the cpu has twice the work to do.

FYI: Ampache 3.8.3 has been released.  A lot of fixes and a few new
features.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug
